husarion_ugv_lights/include/husarion_ugv_lights/led_components/led_segment.hpp

```diff
@@ -128,6 +130,7 @@ class LEDSegment
   std::size_t first_led_iterator_;
   std::size_t last_led_iterator_;
   std::size_t num_led_;
+  std::vector<std::unique_ptr<SegmentLayerInterface>> layers_;
```

Comment:
This could be a map, with priorities defined in enum/struct:
```cpp
enum AnimationPriority {
  PRIORITY_1 = 1,
  PRIORITY_2,
  PRIORITY_3,
  PRIORITY_4,
}
```